NetSDK/Services/SplitFetcher/Classes/InMemorySplitFetcher.cs

```diff
@@ -11,23 +13,20 @@ namespace Splitio.Services.SplitFetcher.Classes
 {
     public class InMemorySplitFetcher: ISplitFetcher
     {
-        protected ConcurrentDictionary<string, ParsedSplit> splits;
-
-        public InMemorySplitFetcher(ConcurrentDictionary<string, ParsedSplit> splits = null)
+        protected ISplitCache splitCache;
+        public InMemorySplitFetcher(ISplitCache splitCache)
         {
-            this.splits = splits ?? new ConcurrentDictionary<string, ParsedSplit>();
+            this.splitCache = splitCache ?? new SplitCache(new ConcurrentDictionary<string, ParsedSplit>());
         }
 
         public ParsedSplit Fetch(string feature)
         {
-            ParsedSplit value;
-            splits.TryGetValue(feature, out value);
-            return value;
+            return splitCache.GetSplit(feature);
         }
 
         public List<ParsedSplit> FetchAll()
```

Comment:
In Java, the manager have access to the fetcher, which hides which cache implementation is being used. My 2 cents.